### PR TITLE
Remove component.yaml

### LINKED ISSUE #9830: [Enhancement]: Enable customize milvus configurations in milvus.yml
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

move configurations to milvus.yml so that users can customize it with helm/operator, specially for the items below:

- [ ] datacoord.segment.maxSize
- [ ] dataNode.flush.insertBufSize
- [ ] knowhere.simdType
- [ ] proxy.maxFieldNum
- [ ] proxy.maxShardNum
- [ ] proxy.maxTaskNum
- [ ] queryNode.segcore/chunkRows  (TBD)
- [ ] rootcoord.dmlChannelNum
- [ ] rootcoord.maxPartitionNum
- [ ] rootcoord.minSegmentSizeToEnableIndex

### Why is this needed?

users are able to customize milvus deployment by updating configurations in milvus.yml

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by xiaofan-luan**
fix #9830
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

File: internal/util/paramtable/basetable.go [hunk 1]
```diff
@@ -128,7 +127,6 @@ func (gp *BaseTable) loadFromCommonYaml() bool {
 		}
 		return true
 	}
-	log.Debug("failed to find common.yaml in config, skip..")
```
Comment (xige-16) [inline internal/util/paramtable/basetable.go:131]: common.yaml has been deleted，is this function work?

Comment (xiaofan-luan) [reply internal/util/paramtable/basetable.go:131]: This won't work, just in case user with old configs put some of the configs into common.yaml

**Pull request merged**